src/libraries/System.Text.Json/source_generation/src/JsonSourceGeneratorHelper.Generate.cs

```diff
@@ -0,0 +1,672 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace System.Text.Json.SourceGeneration
+{
+    internal sealed partial class JsonSourceGeneratorHelper
+    {
+        private readonly string _generationNamespace;
+
+        // TODO: consider public option for this.
+        private bool _honorRuntimeProvidedCustomConverters = true;
```

Comment:
`_honorRuntimeProvidedCustomConverters` is always `true` for now?

Reply:
It's always `true` for now. I think it would be a reasonable feature request to reduce gen'd code and make initialization a bit faster.